x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/ApiKeyServiceTests.java

```diff
@@ -196,16 +196,18 @@ public void testAuthenticateWithApiKey() throws Exception {
 
         final User user;
         if (randomBoolean()) {
-            user = new User("hulk", new String[] { "superuser" }, new User("authenticated_user", new String[] { "other" }));
+            user = new User(new User("hulk", new String[]{"superuser"}, "hulk", "hulk@test.com", Map.of(), true), new User("authenticated_user", new String[]{"other"}));
         } else {
-            user = new User("hulk", new String[] { "superuser" });
+            user = new User("hulk", new String[]{"superuser"}, "hulk", "hulk@test.com", Map.of(), true);
         }
         mockKeyDocument(service, id, key, user);
 
         final AuthenticationResult auth = tryAuthenticate(service, id, key);
         assertThat(auth.getStatus(), is(AuthenticationResult.Status.SUCCESS));
         assertThat(auth.getUser(), notNullValue());
         assertThat(auth.getUser().principal(), is("hulk"));
+        assertThat(auth.getUser().fullName(), is("hulk"));
```

Comment:
nit: use Bruce Baner, or anything else for that matter so that it's easier to spot it's different from the principal, in the eventual case of troubleshooting a test failure

Reply:
++ nice one, fixed.